packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -106,73 +105,76 @@ export class ZkEvmProvider implements Provider {
     }
   };
 
-  private isLoggedIn(): this is LoggedInZkEvmProvider {
+  private isLoggedIn(user?: User | null): this is LoggedInZkEvmProvider {
     return this.magicProvider !== undefined
-      && this.user !== undefined
+      && user !== null
       && this.relayerClient !== undefined
       && this.guardianClient !== undefined;
   }
 
   private async performRequest(request: RequestArguments): Promise<any> {
     switch (request.method) {
       case 'eth_requestAccounts': {
-        if (this.isLoggedIn()) {
-          return [this.user.zkEvm.ethAddress];
+        const existingUser = await this.authManager.getUser() as UserZkEvm;
+        if (this.isLoggedIn(existingUser)) {
+          return [existingUser.zkEvm.ethAddress];
         }
         const { magicProvider, user } = await loginZkEvmUser({
+          existingUser,
           authManager: this.authManager,
           config: this.config,
           magicAdapter: this.magicAdapter,
           multiRollupApiClients: this.multiRollupApiClients,
           jsonRpcProvider: this.jsonRpcProvider,
         });
 
-        this.user = user;
         this.magicProvider = magicProvider;
         this.relayerClient = new RelayerClient({
           config: this.config,
           jsonRpcProvider: this.jsonRpcProvider,
-          user: this.user,
+          authManager: this.authManager,
         });
         this.guardianClient = new GuardianClient({
           confirmationScreen: this.confirmationScreen,
           config: this.config,
+          authManager: this.authManager,
         });
 
-        this.eventEmitter.emit(ProviderEvent.ACCOUNTS_CHANGED, [this.user.zkEvm.ethAddress]);
+        this.eventEmitter.emit(ProviderEvent.ACCOUNTS_CHANGED, [user.zkEvm.ethAddress]);
 
-        return [this.user.zkEvm.ethAddress];
+        return [user.zkEvm.ethAddress];
       }
       case 'eth_sendTransaction': {
+        const user = await this.authManager.getUser() as UserZkEvm;
         if (!this.isLoggedIn()) {
           throw new JsonRpcError(ProviderErrorCode.UNAUTHORIZED, 'Unauthorised - call eth_requestAccounts first');
         }
-
         return sendTransaction({
           params: request.params || [],
           magicProvider: this.magicProvider,
           guardianClient: this.guardianClient,
           jsonRpcProvider: this.jsonRpcProvider,
           relayerClient: this.relayerClient,
-          user: this.user,
+          user,
         });
       }
       case 'eth_accounts': {
-        return this.isLoggedIn() ? [this.user.zkEvm.ethAddress] : [];
+        const user = await this.authManager.getUser() as UserZkEvm;
+        return this.isLoggedIn() ? [user.zkEvm.ethAddress] : [];
       }
       case 'eth_signTypedData':
       case 'eth_signTypedData_v4': {
+        const user = await this.authManager.getUser() as UserZkEvm;
         if (!this.isLoggedIn()) {
```

Comment:
Looks like we've forgotten to pass the `user` to the `isLoggedIn` function here.
To avoid a similar issue in the future, can we store the `user.zkEvm.ethAddress` in a property & reference? That way, we don't have to call `getUser()` from this module and can just pass the `ethAddress` to subsequent modules (e.g. `sendTransaction`), and `isLoggedIn` can be simplified to:
```
  private isLoggedIn(): this is LoggedInZkEvmProvider {
    return this.magicProvider !== undefined
      && this.ethAddress !== undefined
      && this.relayerClient !== undefined
      && this.guardianClient !== undefined;
  }
```